Correct prefixes for vbfy, vscl, vzero, vone, vidt

unknownbrackets opened the pull request:
Also, apply the write mask in interp to single lane ops (would be a weird nop though, so unlikely.)

Note: these changes generally only affect the interpreter.  Broken off of #11948.

-[Unknown]

unknownbrackets committed:
interp: Fix vbfy prefix handling.

unknownbrackets committed:
interp: Correct vscl prefix handling.

unknownbrackets committed:
interp: Generate constants using prefixes.

This way they properly respect negate in the S prefix.

unknownbrackets committed:
interp: Properly apply mask on single lane ops.

When using something like vadd.s, we should still be applying the mask.
Mainly should only matter if masks are set in a conditional, or if games
nop out instructions.

hrydgard commented on Core/MIPS/MIPSIntVFPU.cpp:
```diff
@@ -348,80 +348,70 @@ namespace MIPSInt
+		int offmask = sz == V_Quad || sz == V_Triple ? 3 : 1;
+		int off = vd & offmask;
+		u32 sprefixRemove = VFPU_ANY_SWIZZLE();
+		u32 sprefixAdd = VFPU_CONST(1, 1, 1, 1) | VFPU_SWIZZLE(off == 0, off == 1, off == (2 & offmask), off == (3 & offmask));
```
I have to admit I'm a little bit confused here. We generate four ones, and swizzle them?

unknownbrackets replied:
This is for identity.  I could make the macros more clear, though.

`VFPU_CONST(1, 1, 1, 1)` just means set the constant flag to 1 for each of the lanes.  It doesn't specify the value of the constant.

`VFPU_SWIZZLE(...)` and `VFPU_ABS()` values are actually used to select the constant (when the flag is set.)

I'll create a separate macro for the const values.

-[Unknown]

unknownbrackets replied:
Updated, I think I would've had trouble understanding this code too in 6 months, the macros should make it clearer (used a separate commit since it changes other ops that already used the old way.)

-[Unknown]

hrydgard replied:
Oh duh, I forgot that the CONST flag just switched the mode. I appreciate the new macros, they indeed make things a lot more clearer.

unknownbrackets committed:
interp: Use a helper to generate prefix consts.

This makes more logical sense that using the VFPU_SWIZZLE and VFPU_ABS
macros to select the constant, although that's how the bits work.
